examples/java/src/main/java/org/apache/beam/examples/complete/cdap/PluginConfigInstantiationUtils.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.examples.complete.cdap;
+
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.plugin.PluginConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Class for getting any filled {@link io.cdap.cdap.api.plugin.PluginConfig} configuration object
+ */
+public class PluginConfigInstantiationUtils {
+
+  private static final Logger LOG = LoggerFactory.getLogger(PluginConfigInstantiationUtils.class);
+
+  /**
+   * @param params map of config fields, where key is the name of the field, value must be String or boxed primitive
+   * @return Config object for given map of arguments and configuration class
+   */
+  public static <T extends PluginConfig> T getPluginConfig(Map<String, Object> params, Class<T> configClass) {
```

Comment:
Is there an easier way to get config with all the fields filled properly?

Reply:
At least we can't use GSON and similar libraries because they try to use a no-args constructor, which might be missed in some of `PluginConfig` classes.